sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/ConnectShims.scala

```diff
@@ -0,0 +1,21 @@
+/*
```

Comment:
Shouldn't this class be shimmed and have two versions, one for Spark 4.0.0 and the other for the rest of the shims?

Reply:
Since I added this in sql-plugin-api, I just kept it in one file as we should not be shimming anything in sql-plugin-api. 
For the SparkSession build fix PR-https://github.com/NVIDIA/spark-rapids/pull/12227 I did add shims for Spark-4.0 and rest of them. Please let me know if that's okay.

Reply:
I think the confusion is that Strategy was just an alias to SparkStrategy and upgrade to Spark 4 only affects the alias.  So let us try to simply use SparkStrategy directly.